Adds extension targets for the register screen (block, action, and menu-item)

Profilist opened the pull request:
Resolves https://github.com/shop/issues-retail/issues/16750 and https://github.com/shop/issues-retail/issues/16751

### Background

This PR adds extension targets for the register screen (block, action, and menu-item) as part of the extensible cash management project.

### Solution

Added three new extension targets for the Register screen:

- `pos.cash-tracking-session-details.block.render` - below "Previous sessions" on the content (left) side
- `pos.cash-tracking-session-details.action.render` - modal
- `pos.cash-tracking-session-details.action.menu-item.render` - below the "Settings" button on the actions (right) side

Added corresponding documentation and examples for each target in both TypeScript and React formats. Updated the extension target types and links in the documentation.

![](https://screenshot.click/register-details-targets.png)

### 🎩

- Verify that the specific APIs supported by each new target are suitable
- Verify the documentation is up to standards
- Verify if there's any missing files that need to be added/updated

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

merkoyep commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Shopify/ui-extensions/3143?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3143** <a href="https://app.graphite.dev/github/pr/Shopify/ui-extensions/3143?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/Shopify/ui-extensions/3143?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#3216** <a href="https://app.graphite.dev/github/pr/Shopify/ui-extensions/3216?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>: 1 other dependent PR ([#3248](https://github.com/Shopify/ui-extensions/pull/3248) <a href="https://app.graphite.dev/github/pr/Shopify/ui-extensions/3248?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>)
* `graphite-base/3216`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

merkoyep committed:
Add POS cash management register screen targets

Adds extension targets for the register details screen (block, action, and menu-item) as part of the extensible cash management project.
